lib/mincer/server.js

```diff
@@ -80,15 +80,21 @@ function get_fingerprint(pathname) {
 
 // Helper to write the code and end response
 function end(res, code) {
-  res.writeHead(code);
-
   if (code >= 400) {
+    if (res.req && res.req.next) {
```

Comment:
IMHO it worth to add comment what is this condition for. Nobody will remember after 1 month :)

PS. + squash to single commit and i will merge.